### PR TITLE
Documentation generation: PostGIS reference, function references, ToC…

### PR DESCRIPTION
**Pull request by mngr777**
… generation

Work in progress.

* Table of contents generation moved to `generate.py`, links now work on Github;
* "PostGIS Integration" section added;
* Function cross-references ("See also: ...") added for indexing functions.

Generating table of contents and references for both extensions in single script allows to create cross-references between functions, directing users to PostGIS wrapper if it exists.
xref

Currently, a bug (?) in [Text::Markdown](https://metacpan.org/release/BOBTFISH/Text-Markdown-1.000031/source/lib/Text/Markdown.pm#L455) causes ToC generated by doctoc to be removed on PGXN:
no-toc

Since "See also: ..." links can work either on Github or PGXN, I suggest removing ToC generation altogether and instead point users to PGXN for API reference.

### CODE REVIEW

**Comment by zachasme** [comment]
Great work! I agree we should drop the ToC and point to PGXN. Please let me know if you need anything to progress on this.

**Comment by mngr777** [comment]
* ToC generation removed;
* more `h3_postgis` function references;
* API reference link in README points to PGXN.

File: scripts/documentation/generate.py [hunk 1]
```diff
@@ -98,39 +284,23 @@ def create_fun_rettype(self, children):
...
-
-        return Function(name, arguments, returntype[0], returntype[1])
+    @v_args(inline=True)
+    def create_agg_stmt(self, name: str, arguments, *params):
+        return CreateAggregateStmt(name, arguments)
 
     def argument_list(self, children):
         return children
 
-    # -- CREATE COMMENT --------------------------------------------------------
```
Comment (zachasme) [inline scripts/documentation/generate.py:110]: Could we keep these comment headers? I think they were helpful

File: scripts/documentation/generate.py [hunk 2]
```diff
@@ -167,48 +320,69 @@ def OPERATOR(self, name):
...
-Some functions do not need docs, we can signal this by not applying a comment.
-This is internal operator functions for example.
-
-We should parse the sql files, extract the above, and then document all things that
-have comments applied.
-
-We should probably hardcode a list of things that should not be documented, so we can
-fail if new function are undocumented.
-
-"""
```
Comment (zachasme) [inline scripts/documentation/generate.py:192]: I think it is useful to have a the general idea of the script documented in a comment. Could you re-insert a similar comment in the top of the file?

**Review by zachasme — CHANGES_REQUESTED** [review]
Your pull request looks great! My only nit is the removal of comments, would you consider re-applying them?

**Review by zachasme — APPROVED** [review]

**Pull request merged**